cmd/ncproxy/ncproxy.go

```diff
@@ -313,19 +407,27 @@ func (s *grpcService) AddEndpoint(ctx context.Context, req *ncproxygrpc.AddEndpo
 		trace.StringAttribute("endpointName", req.Name),
 		trace.StringAttribute("namespaceID", req.NamespaceID))
 
-	if req.Name == "" {
+	if req.Name == "" || req.NamespaceID == "" {
 		return nil, status.Errorf(codes.InvalidArgument, "received empty field in request: %+v", req)
 	}
 
-	ep, err := hcn.GetEndpointByName(req.Name)
-	if err != nil {
-		if _, ok := err.(hcn.EndpointNotFoundError); ok {
-			return nil, status.Errorf(codes.NotFound, "no endpoint with name `%s` found", req.Name)
+	if endpt, err := s.ncproxyNetworking.GetEndpointByName(ctx, req.Name); err == nil {
+		endpt.NamespaceID = req.NamespaceID
+		if err := s.ncproxyNetworking.UpdateEndpoint(ctx, endpt); err != nil {
+			return nil, errors.Wrapf(err, "failed to update endpoint with name `%s`", req.Name)
+		}
+	} else {
+		// see if this is an hcn endpoint
+		ep, err := hcn.GetEndpointByName(req.Name)
+		if err != nil {
+			if _, ok := err.(hcn.EndpointNotFoundError); ok {
+				return nil, status.Errorf(codes.NotFound, "no endpoint with name `%s` found", req.Name)
+			}
+			return nil, errors.Wrapf(err, "failed to get endpoint with name `%s`", req.Name)
+		}
+		if err := hcn.AddNamespaceEndpoint(req.NamespaceID, ep.Id); err != nil {
```

Comment:
Originally had a comment here but it was vacation brain speaking, free to ignore 😆